.eslintrc

```diff
@@ -14,7 +14,7 @@
   ],
   "rules": {
     "react-hooks/rules-of-hooks": "error",
-    "react-hooks/exhaustive-deps": "error",
+    "react-hooks/exhaustive-deps": "off",
```

Comment:
Actually, eslint forcing to enter everything as  dependencies which we used inside useEffect hook. we need to add dependencies on which we want to re-execute useEffect hook code. If we will add everything as dependency, it will cause the issue of infinite rendering.
See below error for reference:
```
  useEffect(() => {
    window.addEventListener("keydown", handleMenuKeys);
    window.addEventListener("click", handleClickOutside);
    return () => {
      window.removeEventListener("keydown", handleMenuKeys);
      window.removeEventListener("click", handleClickOutside);
    };
  }, [isOpen, menuRef]);
```

`
 320:6  error  React Hook useEffect has missing dependencies: 'handleClickOutside' and 'handleMenuKeys'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
`

I don't think so we really need to pass functions `handleClickOutside` `handleMenuKeys` as dependencies. 

Earlier we tried to fixed  this kind of all eslint issue by adding dependencies but it was causing issue of infinite re-rendering. 

Let me know the better solution. I think we can add rule as `warning` instead of `error`. wdyt ?

Reply:
You have to wrap the missing dependencies with `useCallback`, and specify for them the right dependencies too. `handleMenuKeys` depends on `isOpen` for example.